app/src/main/java/com/quran/labs/androidquran/ui/PagerActivity.java

```diff
@@ -2132,4 +2140,14 @@ public void onPanelExpanded(View panel) {
     public void onPanelAnchored(View panel) {
     }
   }
+
+  private class NoisyAudioStreamReceiver extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+      if (AudioManager.ACTION_AUDIO_BECOMING_NOISY.equals(intent.getAction())) {
```

Comment:
Yes. 

`PagerActivity` is already cluttered with lots of logic. Also, IMHO, handling service logic (audio playback) in an activity is not a good idea. I did not take a close look, but perhaps, this could better fit to `AudioService` instead.